### PR TITLE
Upload dynamic library debug information (llwebrtc.pdb, etc.) to the crash logger in addition to the main binary.

### PR DESCRIPTION
**Pull request by roxanneskelly**
Previously, only the Second Life main executable debug symbols were uploaded as artifacts to the build, hence uploaded to the crash logger.  Now, additional symbols (.dSYM/pdb/etc.) may be included, such as that for llwebrtc.

Mac symbol files (dSYM) are now generated automatically during build (when running a BUGSPLAT build) instead of requiring a separate build step to generate them.  These symbols are gathered in a staging subdirectory (symbols) which is then archived with tar for upload.
Windows pdb files are copied to the symbols staging directory as part of their build, and are archived with tar for upload.

### CODE REVIEW

**Review by marchcat — APPROVED** [review]

**Review by marchcat — APPROVED** [review]

**Review by nat-goodspeed — APPROVED** [review]
I like the simplification of the CMake logic.

I'm less enthused about baking the `symbol-upload` version into `build.yaml`. The simplest change would update the relevant `viewer-build-util` actions. We could try using a repo or organization variable instead if you really want to avoid invoking `viewer-build-util` for symbol upload.

File: indra/llwebrtc/CMakeLists.txt [hunk 1]
```diff
@@ -31,16 +31,27 @@ add_library (llwebrtc SHARED ${llwebrtc_SOURCE_FILES})
 set_target_properties(llwebrtc PROPERTIES PUBLIC_HEADER llwebrtc.h)
 
 if (WINDOWS)
-    target_link_libraries(llwebrtc PRIVATE ll::webrtc 
+    set_target_properties(llwebrtc
+        PROPERTIES
+        LINK_FLAGS "/debug /LARGEADDRESSAWARE"
```
[the remark this answers is not in the document]
Comment (roxanneskelly) [reply indra/llwebrtc/CMakeLists.txt:36]: We need /debug to generate the debug symbols for upload to BugSplat, so I copied this pattern from the viewer main executable.

File: indra/llwebrtc/CMakeLists.txt [hunk 2]
```diff
@@ -31,16 +31,27 @@ add_library (llwebrtc SHARED ${llwebrtc_SOURCE_FILES})
...
                                        iphlpapi)
+    if (USE_BUGSPLAT)
+        set_target_properties(llwebrtc PROPERTIES PDB_OUTPUT_DIRECTORY "${SYMBOLS_STAGING_DIR}/${VIEWER_CHANNEL}")
+    endif (USE_BUGSPLAT)
 elseif (DARWIN)
     target_link_libraries(llwebrtc PRIVATE ll::webrtc)
+    if (USE_BUGSPLAT)
+        set_target_properties(llwebrtc PROPERTIES XCODE_ATTRIBUTE_DEBUG_INFORMATION_FORMAT "dwarf-with-dsym"
+                                                  XCODE_ATTRIBUTE_DWARF_DSYM_FOLDER_PATH "${SYMBOLS_STAGING_DIR}/${VIEWER_CHANNEL}/dSYMs")
+    endif (USE_BUGSPLAT)
```
[the remark this answers is not in the document]
Comment (roxanneskelly) [reply indra/llwebrtc/CMakeLists.txt:54]: I just updated the definition of SYMBOLS_STAGING_DIR in Linking.cmake as symbols are created during linking.

File: indra/newview/CMakeLists.txt [hunk 3]
```diff
@@ -2159,73 +2157,51 @@ if (INSTALL)
...
-        COMMAND rm -rf "${VIEWER_APP_DSYM}"
-        COMMAND touch "${CMAKE_CURRENT_BINARY_DIR}/dsym.stamp"
-        DEPENDS "${VIEWER_APP_XCARCHIVE}"
-        COMMENT "Cleaning up dSYM"
-        )
+          COMMAND "zip"
+          ARGS
+            "-r"
+            "${VIEWER_APP_XCARCHIVE}"
+            "${product}"
```
[the remark this answers is not in the document]
Comment (roxanneskelly) [reply indra/newview/CMakeLists.txt:2197]: fixed it to be VIEWER_CHANNEL

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -373,15 +373,32 @@ jobs:
...
+      - name: Download Windows Symbols
+        if: env.BUGSPLAT_USER && env.BUGSPLAT_PASS
+        uses: actions/download-artifact@v4
+        with:
+          name: Windows-symbols
+      - name: Extract viewer pdb
+        if: env.BUGSPLAT_USER && env.BUGSPLAT_PASS
+        shell: bash
+        run: |
+          tar -xJf "${{ needs.build.outputs.viewer_channel }}.sym.tar.xz" -C _artifacts
```
[the remark this answers is not in the document]
Comment (roxanneskelly) [reply .github/workflows/build.yaml:390]: Updated the fork of Bugsplat-Git/symbol-upload in secondlife-3p/symbol-upload to the current version from Bugsplat-Git.  Set a vMAJOR tag, which we can change to point to whatever is solid at the time.  This gets around the need to have an intermediate post-bugsplat-windows action.

**Pull request merged**